src/main/java/org/truffleruby/core/rope/RopeCache.java

```diff
@@ -78,6 +78,10 @@ public boolean contains(Rope rope) {
         return bytesToRope.get(key) != null;
     }
 
+    public void rehash() {
+        bytesToRope = new WeakValueCache<>(bytesToRope);
```

Comment:
I think this would look more natural with a `WeakValueCache#rehash` method. Then it would look like `bytesToRope = bytesToRope.rehash();`

Reply:
That approach cannot work for the ConcurrentHashMap cases which need to be re-hashed and the only way is via the copy constructor afaik, so I followed the Java approach here.
But I'll change, as less classes need to know about rehashing.